lib/fmrest/spyke/relation.rb

```diff
@@ -204,18 +204,15 @@ def query(*params)
         with_clone do |r|
           params = params.flatten.map { |p| normalize_query_params(p) }
 
-          if r.or_flag || r.query_params.empty?
+          if r.or_flag || r.query_params.empty? # OR
             r.query_params += params
             r.or_flag = nil
-          elsif params.length > r.query_params.length
-            params[0, r.query_params.length].each_with_index do |p, i|
-              r.query_params[i].merge!(p)
-            end
-
-            remainder = params.length - r.query_params.length
-            r.query_params += params[-remainder, remainder]
-          else
-            params.each_with_index { |p, i| r.query_params[i].merge!(p) }
+          else # AND
+            r.query_params = r
+              .query_params
+              .product(params)
+              .map { |a, b| a.merge(b) { |_, v1, v2| v1 == v2 ? v1 : nil } }
```

Comment:
maybe this should be `v1.to_s == v2.to_s` in case `v1` is `3` and `v2` is `"3"`?